### PR TITLE
[Doc Improvement][Teams AI library code sample update][3789693]

### PR DESCRIPTION
**Pull request by v-sdhakshina**

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md](https://github.com/MicrosoftDocs/msteams-docs/blob/466dfeb95393655d379acec13dbd88d772d9dee9/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fhow-conversation-ai-core-capabilities.md) | 90 | 13 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/7f4b3b05-2e1a-43ef-be06-c94919b66e0b) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/bots/how-to/Teams conversational AI/teams-conversation-ai-overview.md](https://github.com/MicrosoftDocs/msteams-docs/blob/466dfeb95393655d379acec13dbd88d772d9dee9/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fteams-conversation-ai-overview.md) | 91 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/daebd3b1-bccb-4375-b411-75aeb25e40e1) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d6a83b59856fec5587a2cc9126113f65b8685b12/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fhow-conversation-ai-core-capabilities.md) | 90 | 13 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/252c507c-fd51-4615-b317-c25d71e12e5e) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/bots/how-to/Teams conversational AI/teams-conversation-ai-overview.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d6a83b59856fec5587a2cc9126113f65b8685b12/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fteams-conversation-ai-overview.md) | 91 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b8e5faef-af3d-420e-aa4f-d066e4cf4688) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md [hunk 1]
```diff
@@ -196,17 +196,25 @@ app.messageExtensions.selectItem(async (context: TurnContext, state: TurnState,
...
+    query_dict = query.as_dict()
+    search_query = ""
+    if query_dict["parameters"] is not None and len(query_dict["parameters"]) > 0:
+        for parameter in query_dict["parameters"]:
+            if parameter["name"] == "queryText":
+                search_query = parameter["value"]
+                break
+    count = query_dict["query_options"]["count"] if query_dict["query_options"]["count"] else 10
+    url = "http://registry.npmjs.com/-/v1/search?"
+    params = {"size": count, "text": search_query}
```
Comment (lilyydu) [inline msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:217]: missing extra code (need to click "Show all lines" in Loop component)

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md [hunk 2]
```diff
@@ -274,17 +282,24 @@ app.adaptiveCards.actionSubmit('StaticSubmit', async (context, _state, data: Sub
...
 
-# Pass a function to this method
-app.adaptive_cards.action_submit("submit")(execute_submit)
+@app.adaptive_cards.action_submit("StaticSubmit")
+async def on_static_submit(context: TurnContext, _state: AppTurnState, data) -> None:
+    await context.send_activity(f'Statically selected option is: {data["choiceSelect"]}')
+
+@app.adaptive_cards.action_submit("DynamicSubmit")
+async def on_dynamic_submit(context: TurnContext, _state: AppTurnState, data) -> None:
+    await context.send_activity(f'Dynamically selected option is: {data["choiceSelect"]}')
```
Comment (lilyydu) [inline msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:302]: Missing extra code here as well from Loop component

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md [hunk 3]
```diff
@@ -626,16 +641,25 @@ app.messageExtensions.selectItem(async (context, state, item) => {
...
+    query_dict = query.as_dict()
+    search_query = ""
+    if query_dict["parameters"] is not None and len(query_dict["parameters"]) > 0:
+        for parameter in query_dict["parameters"]:
+            if parameter["name"] == "queryText":
+                search_query = parameter["value"]
+                break
+    count = query_dict["query_options"]["count"] if query_dict["query_options"]["count"] else 10
+    url = "http://registry.npmjs.com/-/v1/search?"
+    params = {"size": count, "text": search_query}
```
Comment (lilyydu) [inline msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:662]: Missing code here as well

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md [hunk 4]
```diff
@@ -781,6 +805,31 @@ All entities are required parameters to actions.
...
+    state.ensure_list_exists(parameters.list)
+    items = state.conversation.lists[parameters.list]
+    if parameters.items is not None:
+        for item in parameters.items:
+            items.append(item)
+        state.conversation.lists[parameters.list] = items
+    return "items added. think about your next action"
+
+@app.ai.action("removeItems")
+```
```
Comment (lilyydu) [inline msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:831]: Missing code

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md](https://github.com/MicrosoftDocs/msteams-docs/blob/fc33087ade4322e5e3bb07114aad9423a13ab9b4/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fhow-conversation-ai-core-capabilities.md) | 90 | 13 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3e111b0b-3040-4f14-bac4-6723f6f77287) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/bots/how-to/Teams conversational AI/teams-conversation-ai-overview.md](https://github.com/MicrosoftDocs/msteams-docs/blob/fc33087ade4322e5e3bb07114aad9423a13ab9b4/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fteams-conversation-ai-overview.md) | 91 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/485841f3-cc84-4160-a395-61575091ec2c) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md (hunk 4 above)
Comment (v-sdhakshina) [reply msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:884]: Updated the complete code snippet.

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md (hunk 3 above)
Comment (v-sdhakshina) [reply msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:681]: Updated the complete code snippet.

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md (hunk 2 above)
Comment (v-sdhakshina) [reply msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:315]: Updated the complete code snippet.

File: msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md (hunk 1 above)
Comment (v-sdhakshina) [reply msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md:217]: Updated the complete code snippet.

**Review by lilyydu — APPROVED** [review]

**Review by aycabas — APPROVED** [review]

**Review by govil1811 — APPROVED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md](https://github.com/MicrosoftDocs/msteams-docs/blob/f7386eaab0393b4bec48b2add25d589b40378c7b/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fhow-conversation-ai-core-capabilities.md) | 90 | 12 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/2f127dca-1acc-49f4-91fe-41be21e990c8) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/bots/how-to/Teams conversational AI/teams-conversation-ai-overview.md](https://github.com/MicrosoftDocs/msteams-docs/blob/f7386eaab0393b4bec48b2add25d589b40378c7b/msteams-platform%2Fbots%2Fhow-to%2FTeams%20conversational%20AI%2Fteams-conversation-ai-overview.md) | 91 | 35 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/cded9fca-77ae-4dd9-9190-9e54ef274f61) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Pull request merged**